### DISCUSSION TITLE
Monitor DNS packet data in a specific format

### DISCUSSION THREAD

**Discussion by jinfosec** [zeek/zeek#2838] [Help] [answered]
Hello, 
I'm trying to archive something that I'm not really able to determine if its possible or not by reading the docs. 
I want to be able to process only DNS packet data. PCAP is not an option because I don't have information about all the packet, only the DNS part. 
My first question is:  **It's possible to archive this functionality?** 

To add some complexity to the problem, the rest of the packet information is in a human readable format: Timestamp and other 4th layer information. 
My second question is: **How can I make a simple parser to read this information and combine it with the DNS packet data?**

This pipeline will not be processed in real-time. I'm just trying to come up with a way to generate Zeek events based on DNS and a restricted set of characteristics of an UDP connection (I'm not interested in other scenarios where TCP is used for zone transfers).
Thank you for your time!

**Comment by awelzel** [comment, marked as answer]
> To add some complexity to the problem, the rest of the packet information is in a human readable format: Timestamp and other 4th layer information.
> My second question is: How can I make a simple parser to read this information and combine it with the DNS packet data?

This is putting things a bit on its head: Could you take the human readable format you have and reconstruct DNS packets from it and generate a pcap? It sounds you might have enough data available (or could put placeholder values for other things you're not interested in). In other words, a "your-custom-format-to-pcap" tool with Python scapy or so might be easier to build than extending Zeek directly. It would allow you to use other pcap tools as well.

Is the format your dealing with proprietary or open?

**Reply by jinfosec** [reply]
Thank you for your feedback, somehow I thought that saying Zeek to only parse a specific protocol, in my case DNS would be easier. The format is Microsoft DNS Analytic events.
